python_modules/libraries/dagster-k8s/dagster_k8s/client.py

```diff
@@ -730,14 +730,25 @@ def _has_container_logs(self, container_status):
 
         return False
 
-    def get_pod_debug_info(self, pod_name, namespace, container_name: Optional[str] = None) -> str:
-        pods = self.core_api.list_namespaced_pod(
-            namespace=namespace, field_selector="metadata.name=%s" % pod_name
-        ).items
-        pod = pods[0] if pods else None
+    def get_pod_debug_info(
+        self,
+        pod_name,
+        namespace,
+        container_name: Optional[str] = None,
+        pod: Optional[kubernetes.client.V1Pod] = None,  # the already fetched pod
+    ) -> str:
+        if pod is None:
+            pods = self.core_api.list_namespaced_pod(
+                namespace=namespace, field_selector="metadata.name=%s" % pod_name
+            ).items
+            pod = pods[0] if pods else None
 
         pod_status_str = self._get_pod_status_str(pod) if pod else f"Could not find pod {pod_name}"
 
+        if container_name is None and pod is not None and pod.spec and pod.spec.containers:
+            # assume the first container is relevant if explicit container name not selected
```

Comment:
similar comment to the above. should we be checking an invariant that there is only 1 container?

Reply:
the vast majority of the time its only one container

We allow users to have multiple containers if they pass a base pod spec, and the currently chosen scheme for "which container is the one doing the real work" is to assume its the first position which we call out in the API docs. We can change that scheme before we leave experimental but it seemed reasonable enough to start with.